.github/workflows/scripts/team-approval-check.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-env node */
+
+module.exports = {
+  check: async ({github, context, core, team_slug}) => {
+    const owner = context.repo.owner;
+    const repo = context.repo.repo;
+    const pull_number = context.issue.number;
+
+    // Get all reviews for this PR
+    const {data: reviews} = await github.rest.pulls.listReviews({
+      owner,
+      repo,
+      pull_number,
+    });
+
+    // If no reviews are found, fail early (reduces snumber of API calls)
+    if (reviews.length === 0) {
+      core.setFailed('No reviews found for this PR');
+      return;
+    }
+
+    // Collect all members of the team
+    const {data: members} = await github.rest.teams.listMembersInOrg({
+      org: owner,
+      team_slug,
+    });
+    if (!Array.isArray(members) || members.length === 0) {
+      core.setFailed(`No members found in ${team_slug}`);
+      return;
+    }
+    const memberIds = members.map(member => member.id);
```

Comment:
If you convert this to a set, it will be a lot easier and faster to check with .has()